### ISSUE TITLE
Test toolshed cross origin script issues

### ISSUE THREAD

**Issue by dannon** [galaxyproject/galaxy#7132]
Test toolshed usage with the standard (old) installer from Galaxy is currently broken (pretty sure due to #7005, but I haven't verified for sure, yet).  The beta installer doesn't have this problem since it just uses the API from the toolshed and none of the interfaces.

One fix (from the client side) would be to set allow cross origin headers on all the requests for configured toolshed domains, but that doesn't help all the broken galaxies.  We should be able to fix it from the toolshed side as well, by changing the way we create the singleton Galaxy object there, but it'll take a little work.

Here's an example error:
```
singleton.js?4a51:51 Uncaught DOMException: Blocked a frame with origin "http://127.0.0.1:9009" from accessing a cross-origin frame.
    at getGalaxyInstance (webpack-internal:///./galaxy/scripts/app/singleton.js:55:18)
    at eval (webpack-internal:///./galaxy/scripts/app/monitor.js:27:85)
    at Module../galaxy/scripts/app/monitor.js (http://127.0.0.1:9009/static/scripts/bundled/libs.chunk.js?v=1545229702:47:1)
    at __webpack_require__ (http://127.0.0.1:9009/static/scripts/bundled/extended.bundled.js?v=1545229702:79:30)
    at eval (webpack-internal:///./galaxy/scripts/app/index.js:7:66)
    at Module../galaxy/scripts/app/index.js (http://127.0.0.1:9009/static/scripts/bundled/libs.chunk.js?v=1545229702:35:1)
    at __webpack_require__ (http://127.0.0.1:9009/static/scripts/bundled/extended.bundled.js?v=1545229702:79:30)
    at eval (webpack-internal:///./galaxy/scripts/onload/index.js:8:61)
    at Module../galaxy/scripts/onload/index.js (http://127.0.0.1:9009/static/scripts/bundled/libs.chunk.js?v=1545229702:3181:1)
    at __webpack_require__ (http://127.0.0.1:9009/static/scripts/bundled/extended.bundled.js?v=1545229702:79:30)
```

**Comment by martenson** [comment]
Is this dev Galaxy trying to work with dev toolshed, or is it specific to TTS?

**Comment by dannon** [comment]
Any galaxy trying to get to TTS (or any other dev toolshed without cross origin workarounds).  I suggested a few options to @Nerdinacan and we think we should be able to fix it quickly and redeploy to TTS, fixing it from that side.

**Comment by Nerdinacan** [comment]
I got it. Gimme 10 mins and some coffee.

**Comment by martenson** [comment]
I also encountered a console error when visiting repository page, e.g. https://testtoolshed.g2.bx.psu.edu/view/iuc/abyss/51ca82f3ffaf
```
Uncaught TypeError: Cannot read property 'get' of undefined
    at restore_folder_state (manage_repository?id=2e1e4b2e1914fd19:52)
```
which prevents all repo page interactions.

**Comment by martenson** [comment]
the broken toolshed menu can also be related, supposed to be loading like this: `background: #C1C9E5 url(${h.url_for('/static/style/menu_bg.png')}) top repeat-x;` but results in 404 from `https://testtoolshed.g2.bx.psu.edu/static/style/menu_bg.png`

the file exists at `./client/galaxy/images/menu_bg.png`

edit: this one is probably related to the style changes though

**Comment by Nerdinacan** [comment]
See:
https://github.com/galaxyproject/galaxy/pull/7135

**Comment by jmchilton** [comment]
Is this all fixed with #7146 merged?

**Comment by dannon** [comment]
@jmchilton Yep.  Fixed and deployed (before the holidays -- just forgot to close it out)
